src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -916,6 +898,202 @@ I1<I2> M(A a)
         }
 
         [Fact]
+        public void RefAssemblyClient_EmitTupleNames()
+        {
+            VerifyRefAssemblyClient(@"
+public class A
+{
+    public (int first, int) field;
+}",
+@"class C
+{
+    void M(A a)
+    {
+        System.Console.Write(a.field.first);
+    }
+}",
+comp => comp.VerifyDiagnostics());
+        }
+
+        [Fact]
+        public void RefAssemblyClient_EmitDynamic()
+        {
+            VerifyRefAssemblyClient(@"
+public class A
+{
+    public dynamic field;
+}",
+@"class C
+{
+    void M(A a)
+    {
+        System.Console.Write(a.field.DynamicMethod());
+    }
+}",
+comp => comp.VerifyDiagnostics());
+        }
+
+        [Fact]
+        public void RefAssemblyClient_EmitOut()
+        {
+            VerifyRefAssemblyClient(@"
+public class A
+{
+    public void M(out int x) { x = 1; }
+}",
+@"class C
+{
+    void M(A a)
+    {
+        a.M(out int x);
+    }
+}",
+comp => comp.VerifyDiagnostics());
+        }
+
+        [Fact]
+        public void RefAssemblyClient_EmitVariance()
+        {
+            VerifyRefAssemblyClient(@"
+public interface I<out T>
+{
+}",
+@"
+class Base { }
+class Derived : Base
+{
+    I<Derived> M(I<Base> x)
+    {
+        return x;
+    }
+}",
+comp => comp.VerifyDiagnostics(
+                // (7,16): error CS0266: Cannot implicitly convert type 'I<Base>' to 'I<Derived>'. An explicit conversion exists (are you missing a cast?)
```

Comment:
This looks like failure rather than success. Please either change `out` to `in`, or swap the argument and return types in M.